cocos/scene-graph/node.js

```diff
@@ -1,3046 +1,399 @@
-            for (; i < events.length; ++i) {
-                if (node._bubblingListeners.hasEventListener(events[i])) {
-                    return true;
-                }
-            }
-        }
-        if (node._capturingListeners) {
-            for (; i < events.length; ++i) {
-                if (node._capturingListeners.hasEventListener(events[i])) {
-                    return true;
-                }
-            }
-        }
-        return false;
-    }
-    return true;
-}
-
-function _doDispatchEvent (owner, event) {
-    var target, i;
-    event.target = owner;
-
-    // Event.CAPTURING_PHASE
-    _cachedArray.length = 0;
-    owner._getCapturingTargets(event.type, _cachedArray);
-    // capturing
-    event.eventPhase = 1;
-    for (i = _cachedArray.length - 1; i >= 0; --i) {
-        target = _cachedArray[i];
-        if (target._capturingListeners) {
-            event.currentTarget = target;
-            // fire event
-            target._capturingListeners.emit(event.type, event, _cachedArray);
-            // check if propagation stopped
-            if (event._propagationStopped) {
-                _cachedArray.length = 0;
-                return;
-            }
-        }
-    }
-    _cachedArray.length = 0;
-
-    // Event.AT_TARGET
-    // checks if destroyed in capturing callbacks
-    event.eventPhase = 2;
-    event.currentTarget = owner;
-    if (owner._capturingListeners) {
-        owner._capturingListeners.emit(event.type, event);
-    }
-    if (!event._propagationImmediateStopped && owner._bubblingListeners) {
-        owner._bubblingListeners.emit(event.type, event);
-    }
-
-    if (!event._propagationStopped && event.bubbles) {
-        // Event.BUBBLING_PHASE
-        owner._getBubblingTargets(event.type, _cachedArray);
-        // propagate
-        event.eventPhase = 3;
-        for (i = 0; i < _cachedArray.length; ++i) {
-            target = _cachedArray[i];
-            if (target._bubblingListeners) {
-                event.currentTarget = target;
-                // fire event
-                target._bubblingListeners.emit(event.type, event);
-                // check if propagation stopped
-                if (event._propagationStopped) {
-                    _cachedArray.length = 0;
-                    return;
-                }
-            }
-        }
-    }
-    _cachedArray.length = 0;
-}
+let v3_a = cc.v3();
+let q_a = cc.quat();
+let array_a = new Array(10);
 
-if (CC_EDITOR) {
-    var _oldPosition = new cc.Vec2();
-}
-/**
- * !#en
- * Class of all entities in Cocos Creator scenes.<br/>
- * For events supported by Node, please refer to {{#crossLink "Node.EventType"}}{{/crossLink}}
- * !#zh
- * Cocos Creator 场景中的所有节点类。<br/>
- * 支持的节点事件，请参阅 {{#crossLink "Node.EventType"}}{{/crossLink}}。
- * @class Node
- * @extends _BaseNode
- */
- @ccclass("cc.Node")
+@ccclass('cc.Node')
+@mixins(EventTarget)
 export default class Node extends BaseNode {
+    // local transform
```

Comment:
seems some more info needs to figure out, property is serialized, a cache, and could be synced to editor or not

Reply:
for now all members are preceded with `_`, which makes them private and will not be shown on the editor. more specific details about what and how to expose to the editor could be discuss late after the integration, perhaps?